### PR TITLE
Refactoring CI regarding ARM binary publish

### PR DESCRIPTION
**Pull request by curquiza**
Fixes https://github.com/meilisearch/meilisearch/issues/1909

- Remove CI file to publish aarch64 binary and put the logic into `publish-binary.yml`
- Remove the job to publish armv8 binary
- Fix download-latest script accordingly
- Adapt dowload-latest with the specific case of the MacOS m1

### CODE REVIEW

File: .github/workflows/publish-binaries.yml [hunk 1]
```diff
@@ -38,28 +38,69 @@ jobs:
...
+        if: matrix.use-cross
+        run: |
+          sudo apt-get install -y ${{ matrix.linker }}
+
+      - name: Configure target aarch64 GNU
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+        ## Environment variable is not passed using env:
+        ## LD gold won't work with MUSL
+        # env:
+        #   JEMALLOC_SYS_WITH_LG_PAGE: 16
```
Comment (irevoire) [inline .github/workflows/publish-binaries.yml:82]: Removing that line for the ARM build won’t be a problem?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Whoops, it doesn't work on my Macbook Pro M1.
Capture d’écran 2022-02-07 à 13 23 20

File: download-latest.sh [hunk 2]
```diff
@@ -148,11 +148,18 @@ get_os() {
 get_archi() {
     architecture=$(uname -m)
     case "$architecture" in
-    'x86_64' | 'amd64' | 'arm64')
+    'x86_64' | 'amd64' )
         archi='amd64'
         ;;
+    'arm64')
+        if [ $os -eq 'macos' ]; then # MacOS M1
```
Comment (Kerollmops) [inline download-latest.sh:155]:
```suggestion
        if [ $os = 'macos' ]; then # MacOS M1
```

I just checked and it seems like `-eq` is here to compare numbers (?) and `=` is used to compare strings.

**Review by Kerollmops — CHANGES_REQUESTED** [review]
I am not sure but I would like that first command that returns something different than zero should make the script abort or at least we should check that curl doesn't return something different than 0. It would help in not having `Not Found` in the final `meilisearch` file as it can be dangerous, we `chmod u+x` it and a simple bash script can be returned by a middleware or something... We should abort before `chmod`ing the file.

File: .github/workflows/publish-binaries.yml (hunk 1 above)
Comment (curquiza) [reply .github/workflows/publish-binaries.yml:82]: Which lines? These lines were already commented in the other file

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Ho and BTW the download binary is not the aarch64 one when I use the script from my MacBook Pro M1, it is the `x86_64` one and uses rosetta to run the binary.
```
$ file meilisearch
meilisearch: Mach-O 64-bit executable x86_64
```

**Comment by curquiza** [comment]
> Ho and BTW the download binary is not the aarch64 one when I use the script from my MacBook Pro M1, it is the x86_64 one and uses rosetta to run the binary.

This is expected, when using a M1, the downloaded binary should be x86_64. Maybe I'm wrong, but @tamo confirmed the aarch64 binary did not work on M1. Because, if aarch64 works on M1, I can make the script simpler.

**Comment by Kerollmops** [comment]
Indeed, aarch64 can't work on macOS as it is an ELF binary file, not a Mach-O one. If running the `x86_64` binary on the M1 is expected then the only thing that I don't like is that we do not [stop the script on the first encountered error](https://stackoverflow.com/a/3474556).

**Review by Kerollmops — APPROVED** [review]
Thank you. I'm validating this!

File: .github/workflows/publish-binaries.yml [hunk 3]
```diff
@@ -38,28 +38,69 @@ jobs:
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Installing Rust toolchain
+        uses: actions-rs/toolchain@v1
         with:
-          arch: aarch64 # aka ARMv8
-          distro: ubuntu18.04
-          env: |
-            JEMALLOC_SYS_WITH_LG_PAGE: 16
```
Comment (irevoire) [inline .github/workflows/publish-binaries.yml:52]: @curquiza 
> Which lines? These lines were already commented in the other file

It was not commented here, for the arm binary

Comment (irevoire) [reply .github/workflows/publish-binaries.yml:52]: Ah yes sorry I missed it my bad!

**Review by irevoire — APPROVED** [review]

**Comment by curquiza** [comment]
bors merge

**Pull request merged**